ci/do_ci.sh

```diff
@@ -167,6 +183,41 @@ elif [[ "$1" == "check_format" ]]; then
 elif [[ "$1" == "docs" ]]; then
   docs/publish.sh
   exit 0
+elif [[ "$1" == "github_release" ]]; then
+  if [[ ! -f "${ENVOY_SRCDIR}/build_release_stripped/envoy" ]]; then
```

Comment:
Can you move this into a new script `ci/publish_github_release.sh`? There's a fair bit of logic here that would be easier to understand in isolation. In the new script, can you also add a comment at the top explicitly warning not to set `set -x` debug, since we can get dangerous leakage of GH creds via CI logs if we don't take this step. This hit us previously in Docker CI image push, see https://github.com/envoyproxy/envoy/blob/master/ci/build_container/docker_push.sh#L3.